### PR TITLE
Fix BatchNormalization test

### PR DESCRIPTION
**Pull request by disktnk**
- support train mode on exporting
- add train mode test on BatchNormalization
- add train mode test on grad output

TODO

- ~change `chainer.config` only when exporting, revert to previous config after exporting~ done
- support optional output on BatchNormalization op --> to be supported

### CODE REVIEW

File: onnx_chainer/export.py [hunk 1]
```diff
@@ -210,7 +211,11 @@ def export(model, args, filename=None, export_params=True,
 
     _check_available()
 
-    chainer.config.train = False
+    if train:
```
Comment (shinh) [inline onnx_chainer/export.py]: chainer.config.train = train

File: onnx_chainer/export.py [hunk 2]
```diff
@@ -210,7 +211,11 @@ def export(model, args, filename=None, export_params=True,
...
+        chainer.config.train = True
+    else:
+        chainer.config.train = False
+    chainer.config.in_recomputing = False
```
Comment (shinh) [inline onnx_chainer/export.py]: True?

File: tests/functions_tests/test_normalizations.py [hunk 3]
```diff
@@ -68,7 +75,12 @@ def __call__(self, x):
         self.x = input_generator.increasing(2, 5)
 
     def test_output(self):
-        name = 'batchnormalization'
-        if hasattr(self, 'name'):
-            name += '_' + self.name
-        self.expect(self.model, self.x, name=name)
+        train = False
```
Comment (shinh) [inline tests/functions_tests/test_normalizations.py]: train = getattr(self, 'train', False)

File: onnx_chainer/functions/normalization.py [hunk 4]
```diff
@@ -7,23 +7,28 @@
...
-
-    unique_layer_name = '{}_{}'.format(func.__class__.__name__,
-                                       context.get_name(func))
-    num_outputs += [
-        unique_layer_name + '_mean',
-        unique_layer_name + '_var',
-        unique_layer_name + '_saved_mean',
-        unique_layer_name + '_saved_var'
-    ]
+    x = func.inputs[0].get_variable().data
```
Comment (shinh) [inline onnx_chainer/functions/normalization.py]: It'd be nice if we can share the code with FixedBN, but it's up to you

File: onnx_chainer/export.py [hunk 5]
```diff
@@ -97,6 +98,17 @@ def rename_tensors(model):
...
 
+@contextlib.contextmanager
+def set_temporary_chainer_config(**kwargs):
+    org_config = chainer.config
+    temp_conf = chainer.configuration.LocalConfig(chainer.config)
+    for k, v in kwargs.items():
+        setattr(temp_conf, k, v)
+    chainer.config = temp_conf
+    yield
+    chainer.config = org_config
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: y, exactly

File: tests/functions_tests/test_normalizations.py (hunk 3 above)
Comment (disktnk) [reply tests/functions_tests/test_normalizations.py]: fixed

File: onnx_chainer/export.py [hunk 6]
```diff
@@ -97,6 +98,17 @@ def rename_tensors(model):
             v.name = names[v.name]
 
 
+@contextlib.contextmanager
```
[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: `chainer.using_config` is fit for only one key-value pair set, I think, so this PR don't use the method. I couldn't have thought to use Chainer's utility function.

[the remark this answers is not in the document]
Comment (disktnk) [reply onnx_chainer/export.py]: sorry, this code is invalid, `LocalConfig` has thread local so not enough even `chainer.config = tmp`, should use `using_config`.

File: onnx_chainer/functions/normalization.py (hunk 4 above)
Comment (disktnk) [reply onnx_chainer/functions/normalization.py]: fixed, thanks for advice

**Pull request merged**